sphinx_needs/filter_common.py

```diff
@@ -306,9 +306,16 @@ def filter_single_need(
         # Set filter_context as globals and not only locals in eval()!
         # Otherwise, the vars not be accessed in list comprehensions.
         if filter_compiled:
-            result = bool(eval(filter_compiled, filter_context))
+            result = eval(filter_compiled, filter_context)
         else:
-            result = bool(eval(filter_string, filter_context))
+            result = eval(filter_string, filter_context)
+        if isinstance(result, re.Match):
```

Comment:
I think we should do this not here, but in the search function:
https://github.com/useblocks/sphinx-needs/blob/master/sphinx_needs/filter_common.py#L303

I would add something like this to `filter_common.py`:
```python
def need_search(*args, **kwargs)
   return bool(re.search(*args, **kwargs))

... some other code

 filter_context["search"] = need_search
```

Then the result of `search()` gets transformed when used.
And we can save the check for re.Match here, as we should always get boolean-values back.

Sorry for making your PR a little more complex, but you found a new bug with it and for a generic solution for your problem we need to fix this as well.